mmcv/runner/fp16_utils.py

```diff
@@ -186,7 +202,11 @@ def new_func(*args, **kwargs):
                     else:
                         new_kwargs[arg_name] = arg_value
             # apply converted arguments to the decorated method
-            output = old_func(*new_args, **new_kwargs)
+            if TORCH_VERSION != 'parrots' and TORCH_VERSION >= '1.6.0':
+                output = autocast(enabled=False)(old_func)(*new_args,
```

Comment:
It seems that the key difference is to use  `autocast` if we can. So shall we set the case in the header first like
```python
try:
    from torch.cuda.amp import autocast
except ImportError:
    autocast = None
```

Then simply use the condition like `if iscallable(autocast)` here, rather than checking the version of pytorch

Reply:
Actually in previous PyTorch versions (e.g. 1.5), `torch.cuda.amp` is also implemented but we will see warnings that showing the `torch.cuda.amp` is experimental. So using version as the indicator is more reliable here in our case.

Reply:
OK